consensus/core/src/dag_state.rs

```diff
@@ -304,11 +305,12 @@ impl DagState {
         // Ensure we don't write multiple blocks per slot for our own index
         if block_ref.author == self.context.own_index {
             let existing_blocks = self.get_uncommitted_blocks_at_slot(block_ref.into());
-            assert!(
-                existing_blocks.is_empty(),
-                "Block Rejected! Attempted to add block {block:#?} to own slot where \
-                block(s) {existing_blocks:#?} already exists."
-            );
+            if !existing_blocks.is_empty() {
+                debug_fatal!(
```

Comment:
just for my own understanding, how does this help with amnesia recovery? We will log on error here by default now unless the host has debug assertions enabled, which means we will allow for writing of a block into our own slot?

Reply:
Isn't that the correct behavior? We don't want the duplicated own block to be written. If amnesia recovery failed to sync latest blocks to recover own blocks, and we start writing the "newly" generated own blocks, what part of the code will regulate the correct behavior after this?

Reply:
I believe generating an occasional equivocation is ok. But this codepath is not tested that much right now. Ok I will revert this line for now.